csrc/scalar_value.h

```diff
@@ -0,0 +1,224 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <c10/util/BFloat16.h>
+#include <c10/util/Half.h>
+#include <dynamic_type.h>
+#include <complex>
+#include <cstddef>
+#include <unordered_map>
+
+#include <type.h>
+
+namespace nvfuser {
+
+template <typename T>
+struct Struct {
+  // In theory, we should just use std::unordered_map<std::string, T>, but this
+  // doesn't work on old gcc. See also SetTheoreticNaturalNumbers
+#if defined(__clang__) || __GNUC__ >= 12
+  std::unordered_map<std::string, T> fields;
+#define GCC_BUG_Struct
```

Comment:
same here

Reply:
This is actually undefined a few lines after this line.